rclcpp/src/rclcpp/detail/mutex_two_priorities.cpp

```diff
@@ -70,21 +67,25 @@ LowPriorityLockable::lock()
 {
   std::unique_lock<std::mutex> guard{parent_.cv_mutex_};
   while (parent_.data_taken_ || parent_.hp_waiting_count_) {
-    parent_.cv_.wait(guard);
+    parent_.lp_cv_.wait(guard);
   }
   parent_.data_taken_ = true;
 }
 
 void
 LowPriorityLockable::unlock()
 {
+  bool notify_lp{false};
   {
     std::lock_guard<std::mutex> guard{parent_.cv_mutex_};
     parent_.data_taken_ = false;
+    notify_lp = 0u == parent_.hp_waiting_count_;
+  }
+  if (notify_lp) {
```

Comment:
@ivanpauno oh, didn't consider a conditional notification. It seems safe though.

Reply:
Yeah, I really though a lot about that part of the code, but given that `notify_lp` is calculated with the mutex taken this should be safe.